tailnet/controllers.go

```diff
@@ -440,3 +443,83 @@ func (l *derpSetLoop) recvLoop() {
 		l.setter.SetDERPMap(dm)
 	}
 }
+
+type BasicTelemetryController struct {
+	logger slog.Logger
+
+	sync.Mutex
+	client      TelemetryClient
+	unavailable bool
+}
+
+func (b *BasicTelemetryController) New(client TelemetryClient) {
```

Comment:
I'm a little confused by this `TelemetryController` interface now, also having looked at https://github.com/coder/coder/pull/15361. 
Telemetry leaves the conn via this type as `TelemetrySink`, and we otherwise never call `SendTelemetryEvent` from this controller interface. 
What purpose does it serve on the `Controller` struct? Even if we added agent telemetry wouldn't we just use the `TelemetrySink` stored on a `tailnet.Conn` instead?

Reply:
The `TelemetryController` interface is there so that we can handle reconnections to the API.  Each time we reconnect we get a new `TelemetryClient`, so the controller handles routing events to the current client, if available.